freetype: 2.10.0 -> 2.10.1

dtzWill opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#sec-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change

https://sourceforge.net/projects/freetype/files/freetype2/2.10.1/

I've been experimenting with mini bootstrap to get freetype and harfbuzz
built with support for the other, as well as using freetype's default
(instead of "encumbered" cleartype) mostly to see what the state of
things were and for potentially avoiding any/all color fringing
without mucking with lcd filter settings.

This PR simply bumps the freetype version, don't worry :).
Not sure the bootstrap dance helps anything, but if anyone has
**recent** images/data/information on default ("Harmony"?) vs
"encumbered" cleartype that'd be appreciated.

I've been happy with Harmony so far, but maybe I'm not looking at the
right details or use different sorts of programs...

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

dtzWill committed:
freetype: 2.10.0 -> 2.10.1

https://sourceforge.net/projects/freetype/files/freetype2/2.10.1/

ttuegel commented on pkgs/development/libraries/freetype/default.nix:
```diff
@@ -32,20 +32,20 @@ in stdenv.mkDerivation rec {
-    [ ./enable-table-validation.patch
-    ] ++
-    optional useEncumberedCode ./enable-subpixel-rendering.patch;
+    [ ./enable-table-validation.patch ] ++
```
If we are going to reformat this section, I would prefer that we conform to typical formatting for Nixpkgs here, i.e. keeping `optional` (on the next line) indented past `patches` above, and moving `++` onto the next line, ahead of `optional`.

dtzWill committed:
freetype: undo formatting, esp since doesn't move to convention.

Sorry, thanks for catching this in review!

Leaving it as it was to keep this change simpler and
friendlier to blame and such :).

FRidh commented:
@grahamcofborg eval
@GrahamcOfBorg build freetype
